walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,115 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        
+        // no meeting time available 
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return Arrays.asList();
+        }
+        
+        // no other clashing events
+        if (events.isEmpty()) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+
+        // flag to check mutual exclusivity of attendee list
+        boolean distinctAttendees = true;
+
+        List<TimeRange> available = Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true));
+
+        for (Event event : events) {
+            if (attendeesOverlap(event.getAttendees(), request.getAttendees())) {
+                available = remove(available, event.getWhen());
+                distinctAttendees = false;
+            }
+        }
+
+        // mutual exclusivity of attendee list, i.e. no clashing events
+        if (distinctAttendees == true) {
```

Comment:
We don't need to state explicitly == true or == false, we can use
`if (distinctAttendees)`

`if (!distinctAttendees)`

Reply:
Okay got you, I'll definitely keep that in mind for future PRs too.